### ISSUE TITLE
Disabling UAC  popups

### ISSUE THREAD

**Issue by matthussey** [coreybutler/nvm-windows#149]
Hi,

I have found your software and it looks really good for the project I am working on. I am able to use it but one thing I would like to be able to do is run this as part of the build script on my build server but the UAC popups stop this because user intervention is required. I cannot disable UAC on the build server.

I moved my symlink directory out of program files and into a user accessible area, hoping that the UAC popups were because admin was required for changing program files, but this hasn't changed anything.

Do you know of any workarounds? If not, would a possible future feature be to not have UAC if the symlink directory is somewhere that admin is not required?

I have a slight feeling that it's symlink, not the location, that requires UAC but either way the UAC popups do cause me problems for my build server.

Thanks

Matthew

**Comment by damnhandy** [comment]
+1 This is do to the fact that it calls a script the requires admin privileges:

https://github.com/coreybutler/nvm-windows/blob/master/src/nvm.go#L264

It'd be better to not assume elevated privilages and let UAC do it's thing when NVM is writing to a privileged location. If my `NVM_HOME` is within my home directory, it shouldn't require admin rights. 

We are using the [Scoop](https://github.com/lukesampson/scoop) package manager to get around this very issue. However, it only allows the installation of single version of node. So far, NVM-Windows is the best version manager I have found to date, but the admin rights thing is a problem.

**Comment by coreybutler** [comment]
Sorry for the delayed response.

Unfortunately, @matthussey is correct about the symlinks. Windows requires elevated administrative privileges to manage symlinks.

I'm closing this since, as of now, there is no way to disable UAC on Windows. If Microsoft changes that in the future, it will be reflected in this project.

**Comment by edclement** [comment]
Junctions instead of symlink? This is what npm does...

**Comment by coreybutler** [comment]
Junctions fail when the install is on another partition, like `D:\\`. Unless it's new, I'm pretty sure npm does not use junctions (though I haven't verified). It uses CMD files.

**Comment by matthussey** [comment]
Looking at this, and I'm, no expert, Symlinks work to any physical, logical and remote drive. Junctions work to any physical, logical drive on the same computer. Hardlinks only work to the same physical and logical drive.
I think that maybe junctions could work if the link is on the same computer, which in most cases I imagine it would be.
Maybe a config option would allow flexibility.

**Comment by simlg** [comment]
I needed an easy solution to let people select their nodejs version on our build system but I had the UAC popup problem as well. I went around it by having a complete user side installation on my build machines in %APPDATA% (NVM_HOME and NVM_SYMLINK). I also replaced mklink /D by a junction link (/J) and now I can run nvm install and nvm use without any UAC popup. Hope it helps someone out there with the same problem.
